src/plots/cartesian/set_convert.js

```diff
@@ -182,8 +182,9 @@ module.exports = function setConvert(ax) {
             // encounters them, ie all the categories from the
             // first data set, then all the ones from the second
             // that aren't in the first etc.
-            // TODO: sorting options - do the sorting
-            // progressively here as we insert?
+            // it is assumed that this function is being invoked in the
+            // already sorted category order; otherwise there would be
+            // a disconnect between the array and the index returned
```

Comment:
I'll look into it, thanks! Unrelated: I learnt that date axes take ISO strings `"2016-03-29"` as values. When I expressly indicate axis `type = 'category'` it behaves as categories and the axis ticks get rendered as plaintext `"2016-03-29"` rather than nicely formatted date. So I'm not yet grounded in the approach for coercion which was step #2 in your original comment.